src/fastmcp/server/server.py

```diff
@@ -1332,8 +1332,12 @@ def http_app(
                 or self._deprecated_settings.streamable_http_path,
                 event_store=None,
                 auth=self.auth,
-                json_response=self._deprecated_settings.json_response,
-                stateless_http=self._deprecated_settings.stateless_http,
+                json_response=(
+                    json_response or self._deprecated_settings.json_response
```

Comment:
i think if you pass `json_response=False` the value will be ignored and `self._deprecated_settings.json_response` will be used instead

`json_response if json_response is not None else self._deprecated_settings.json_response` would be safer!

Reply:
🤦 of course